### ISSUE TITLE
CPU and memory usage over time causes process to die

### ISSUE THREAD

**Issue by TheWoodlief** [nwjs/nw.js#4108]
Seeing an issue on Windows and Mac computers (haven't seen it on linux yet) where the CPU and memory use goes up over time. Using the javascript profiler always shows that the cpu is in the idle state very close to 100% of the time. Memory usage in the profiler shows about 34mb usually even though a lot more is shown in the activity monitor. There seem to be around 3 or 4 nw.js processes on each system but sometimes the mac will have lots of dead processes not using the CPU. The application seems to die faster on lower spec systems. Attached are some images from Mac and Windows.

Image 1 Mac: About 2 hours in. At 8 hours in the cpu usage was at around 48% crawling up consistently a few percents per each hour.
<img width="753" alt="nwjs helpers" src="https://cloud.githubusercontent.com/assets/5483193/12045953/e71908a4-ae7b-11e5-9e1b-f61a1c84472b.png">

Image 2 Windows:
![pasted_image_at_2015_12_29_07_43_pm_720](https://cloud.githubusercontent.com/assets/5483193/12045987/8ac80932-ae7c-11e5-9328-5705fffb25cd.png)

Image 2  Mac: Memory Usage
<img width="732" alt="screen shot 2015-12-29 at 9 43 18 pm" src="https://cloud.githubusercontent.com/assets/5483193/12045991/954b42ca-ae7c-11e5-8169-18ddd1688b09.png">
##

**Comment by rogerwang** [comment]
Thanks for reporting. What's the version are you using?

**Comment by TheWoodlief** [comment]
We are on version 0.12.3

**Comment by rogerwang** [comment]
Please try with 0.13.0-beta1, in which we rewrite the related part for mac .

**Comment by jssuttles** [comment]
I'm working with @TheWoodlief on this project and we've tried rebuilding the project with 0.13.0-beta2sdk, but we've run into a few snags including the window properties in the manifest, specifically show: false, frame: false, resizing: false, transparent: true, always_on_top: true. (As mentioned in another issue by someone else, show: false doesn't let the app start. There is some combination of the others that creates some mac error output.) When all properties except height and width are removed, the app runs without spitting out errors except for the emit problem that I mentioned in another issue. Once the window properties all work and we get some method of communication between windows, 0.13.0 should work out for us.

As far as our current bugs, we've discovered that the systray icon is actually disappearing on windows machines after staying idle for long periods of time. Two similar windows machines were running our app for a long period of time. The app on one of the machines was used fairly often, while the other was left idle. After the idle one was checked, the systray had disappeared. This has happened many times, and our customers keep thinking that it has died, so they try to start it again. However, there are nw.exe processes running in the background. Our app is a listening app, so when they perform the action that makes the app pop up (sending something to it), the app does pop up, thereby indicating that the systray icon is somehow disappearing (timing out?).

**Comment by jssuttles** [comment]
@TheWoodlief I think that this issue can be closed.
@rogerwang We actually found that we weren't on 0.12.3. We were on 0.12.1. We updated to 0.12.3 and it seems to be a lot more stable. One of the closing issues we were having we found to be because websockets were closing client-side in certain networks after under 30 minutes. We had to start pinging the client every 5 minutes. That's probably not up your alley, but if you know anything about that and know how to fix it, it would be greatly appreciated.
